sound/Music.java

```diff
@@ -187,37 +192,91 @@ public void playChord(Chord chord, int defaultDuration) {
             System.out.println("Problem sleeping?");
         }
     }
-    
+
     /**
      * Play a tune.
-     * @param tune the tune to play
+     * @param tune the tune to play.
+     * @param imp the ImgProvider object.
      */
-    public void playTune(Tune tune) {
+    public void playTune(final Tune tune, final ImgProvider imp) {
+        ImgProvider improvider = imp;
+        int[] pixels = improvider.getPix();
+        int height = improvider.getPixHeight();
+        int width = improvider.getPixWidth();
+        int newRow = width;
```

Comment:
Why is a variable named "newRow" holding an `int` value?  From the name it seems like it would be of type `int[]`.  That it is intialized with "width" is additionally perplexing.

Reply:
I changed this to pixCount as the variable represents the number of pixels that are provided from pixel[] to row[] during each iteration through the while loop.

Reply:
OK.